utils.py

```diff
@@ -58,8 +58,6 @@ def assemble_program(algod_client: algod.AlgodClient, source_code: str) -> bytes
     return base64.b64decode(compile_response["result"])
 
 
-# NOTE: getter_params represents a tuple of three tuples. This utility
-# will be removed once PyTeal integrates the ABI type NamedTuple
 SmartASAConfig = namedtuple(
```

Comment:
Would it be possible to generate this from the `SmartASAConfig` in the asc? To avoid having duplication around

Reply:
Not sure I understand your request. This is using python's `NamedTuple` which is different from `abi.NamedTuple`. 

This is just syntactic sugar for tests -> [example here](https://github.com/algorandlabs/smart-asa/blob/8045ffb38dc5bb0d07b711db2cea9274b6cfae6e/smart_asa_test.py#L2443)

Reply:
Since the definition is mirroring the one in the ASC, I was wondering if we could avoid having to repeat the field names here and just infer them from the `abi.NamedTuple` `SmartASAConfig` (so that, eg, if we change something on the asc definition this is updated automatically). So I guess the gist is can we generate a python `NamedTuple` from an `abi.NamedTuple`?

Reply:
Probably it would be better to use https://docs.python.org/3/library/inspect.html#inspect.get_annotations to get the keys, other than that SGTM. Thanks!